spec/emerald/parser_spec.rb

```diff
@@ -100,4 +100,48 @@
                           Emerald::Number.new(30)])
     end
   end
+
+  describe "string parsing" do
+    it"parses a string ofcharacters" do
+      source = '"Hello"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello"')])
+    end
+    
+    it"parses a string ofcharacters and whitespace" do
+      source = '"Hello this is a string"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello this is a string"')])
+    end
+
+    it "parses a string of characters, whitespace and digits" do
+      source = '"H3llo this is 1 str1ng"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo this is 1 str1ng"')])
+    end
+
+    it "parses a string of characters, whitespace, digits and symbols" do
+      source = '"H3llo, this is 1 str1ng."' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo, this is 1 str1ng."')])
+    end
+
+    it "parses unusual strings that contain a backslash immediately followed by a quotation mark" do
+      source = '"this is a weird string -> \" <- "'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"this is a weird string -> \" <- "')])
+    end
+    
+    it "parses strings that contain a backslash immediately followed by a quotation mark" do
+      source = '"try \"this"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"try \"this"')])
+    end
+
+    it "parses a string with a set of quotation marks are immediately after a backslash" do
+      source = '"Say \"hello\""'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Say \"hello\""')])
+    end
+  end
 end
```

Comment:
Oh I'm a bit confused. So this regex pattern `(["])(?:[^"\\]|\\.)*\1` should work for: 
```
 \"
```
and this:
```
 \\\"
```
but not
```
 \\"
```

However it appears as though the backslashes are automatically added. E.g. you can see it when this one is forced to fail: 
```
source = '"This is a\" strinmg"'
expect(ast).to eq([Emerald::String.new('"This is a\" string"')])
```
Results in:
```
    Failure/Error: expect(ast).to eq([Emerald::String.new('"This is **a\"** string"')])
     
       expected: [#<Emerald::String:0x007fe3030e29a8 @string="\"This is a\\\" string\"">]
            got: [#<Emerald::String:0x007fe3030e2c00 @string="\"This is **a\\\"** strinmg\"">]
```
I'm finding it hard to figure out how I could deal with that